src/ServerlessAutoSwagger.ts

```diff
@@ -1,398 +1,501 @@
-                parameters.push({
-                    name: param,
-                    in: 'path',
-                    required: true,
-                    type: 'string',
-                });
-            });
-        }
-        if ((httpEvent as FullHttpEvent['http']).parameters?.path) {
-            const rawPathParams = (httpEvent as FullHttpEvent['http']).parameters?.path || {};
-            let pathParameters = httpEvent.path.match(/[^{\}]+(?=})/g) || [];
-            Object.entries(rawPathParams).map(([param, required]) => {
-                parameters.push({
-                    name: param,
-                    in: 'path',
-                    required,
-                    type: 'string',
-                });
-                pathParameters = removeStringFromArray(pathParameters, param);
-            });
-
-            pathParameters.map(param => {
-                parameters.push({
-                    name: param,
-                    in: 'path',
-                    required: true,
-                    type: 'string',
-                });
-            });
+  addEndpointsAndLambda = () => {
+    this.serverless.service.functions = {
+      ...this.serverless.service.functions,
+      ...swaggerFunctions(this.serverless),
+    }
+  }
+
+  generatePaths = () => {
+    const functions = this.serverless.service.functions
+    Object.entries(functions).map(([functionName, config]) => {
+      const events = config.events || []
+      events
+        .filter((event) => {
+          if (!((event as HttpEvent).http || (event as HttpApiEvent).httpApi)) {
+            return false
+          }
+
+          const http =
+            (event as HttpEvent).http || (event as HttpApiEvent).httpApi
+
+          if (typeof http === "string") {
+            return false
+          }
+
+          return !http.exclude
+        })
+        .map((event) => {
+          let http =
+            (event as HttpEvent).http || (event as HttpApiEvent).httpApi
+          if (typeof http === "string") {
+            // TODO they're using the shorthand - parse that into object.
+            return
+          }
+
+          let path = http.path
+          if (path[0] !== "/") path = `/${path}`
+
+          if (!this.swagger.paths[path]) {
+            this.swagger.paths[path] = {}
+          }
+
+          const apiKeyName =
+            this.serverless.service.custom?.autoswagger?.apiKeyName
+
+          var security_key_name: { [key: string]: string[] } = {}
+          security_key_name[apiKeyName] = []
+
+          var security_key = [security_key_name] as MethodSecurity[]
+
+          var securityDefinitions: { [key: string]: SecurityDefinition } = {}
+
+          securityDefinitions[apiKeyName] = {
+            type: "apiKey",
+            name: apiKeyName,
+            in: "header",
+          }
+
+          this.swagger.paths[path][http.method] = {
+            summary: http.summary,
+            description: http.description || functionName,
```

Comment:
I'm thinking `functionName` should be the fallback for `summary` instead of description, but I'm curious to hear @SamWSoftware 's opinion on it

Reply:
I struggle to understand what the difference between description and summary is. Either would be fine in my opinion.